boa/src/bytecompiler.rs

```diff
@@ -868,8 +868,8 @@ impl ByteCompiler {
 
         compiler.code_block.params = paramaters.to_owned().into_boxed_slice();
 
-        compiler.emit(Opcode::PushUndefined, &[]);
-        compiler.emit(Opcode::Return, &[]);
+        // compiler.emit(Opcode::PushUndefined, &[]);
+        // compiler.emit(Opcode::Return, &[]);
```

Comment:
we can't do this without knowing if the last node is a return statement. otherwise the function will never return.

Reply:
I've added it back